libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -206,7 +206,11 @@ void CompilerUtils::abiDecode(TypePointers const& _typeParameters, bool _fromMem
 		}
 		// stack: v1 v2 ... v(k-1) v(k) base_offset mem_offset
```

Comment:
@chriseth I assume `base_offset` was the starting offset and `mem_offset` is where we ended up?

Reply:
No, `base_offset` is the position of the beginning of the current "block" (the point from which on data pointers are taken) and `mem_offset` is the point inside the current block where we are currently reading.

Reply:
Do we have the pointer in memory or is the free memory pointer updated properly?